ts/parts-3d/Column.ts

```diff
@@ -503,22 +494,41 @@ wrap(Series.prototype, 'alignDataLabel', function (
     proceed: Function
 ): void {
 
+    // In 3D we need to pass point.outsidePlot option to the justifyDataLabel
+    // method for disabling justifying dataLabels in columns outside plot
+    arguments[3].outside3dPlot = arguments[1].outside3dPlot;
+
     // Only do this for 3D columns and it's derived series
     if (
         this.chart.is3d() &&
         this instanceof seriesTypes.column
     ) {
-        var series = this as Highcharts.ColumnSeries,
-            chart = series.chart;
-
         var args = arguments,
             alignTo = args[4],
-            point = args[1];
+            point = args[1],
+            dLOptions = args[3];
 
-        var pos = ({ x: alignTo.x, y: alignTo.y, z: series.z });
+        var series = this as Highcharts.ColumnSeries,
+            seriesOptions: Highcharts.ColumnSeriesOptions = series.options,
+            chart = series.chart,
+            inside = pick(dLOptions.inside, !!series.options.stacking),
+            pos = ({
+                x: alignTo.x + point.pointWidth / 2,
+                y: alignTo.y,
+                z: series.z + (seriesOptions as any).depth / 2
+            });
```

Comment:
is the `()` a leftover from testing some TS castings? `pos = ({...})` should be the same as `pos = {...}`